src/snowcli/cli/project/config.py

```diff
@@ -0,0 +1,50 @@
+from pathlib import Path
+from strictyaml import Map, Seq, Optional, UniqueSeq, Bool, Int, Str, Regex, YAMLValidationError, load
+from collections import OrderedDict
+
+IDENTIFIER = "[A-Za-z0-9_]+"
+SCHEMA_AND_NAME = f"{IDENTIFIER}.{IDENTIFIER}"
+
+# TODO: turn into regexes
+Path = Str
+Glob = Str
+
+PathMapping = Map({
+    "src": Glob() | Seq(Glob()),
+    Optional("dest"): Path(),
+})
+
+app_schema = Map({
+    "name": Str(),
+    Optional("deploy_root", default="output/deploy/"): Path(),
+    Optional("source_stage", default="app_src.stage"): Regex(SCHEMA_AND_NAME),
+    "scripts": Map({
+        Optional("package", default="package/*.sql"): Glob() | UniqueSeq(Glob()),
+    }),
+    "artifacts": Seq(Path() | PathMapping)
+})
+
+project_schema = Map({
+    "native_app": app_schema,
+})
+
+local_schema = Map({
+    "package": Map({
+        "role": Regex(IDENTIFIER),
```

Comment:
Why do we need a regex check here? I do understand that snowflake has a requirement around it: https://docs.snowflake.com/en/sql-reference/identifiers-syntax
But should we be performing this check ourselves instead of just passing it on to snowflake, and returning whatever error message (or lack of) that snowflake produces?

Reply:
I don't see these identifier requirements changing anytime, and this follows the maxim of failing fast.